test/api/test_arrow.cpp

```diff
@@ -72,14 +72,10 @@ class MyPythonTableArrowArrayStreamFactory {
 	    : arrow_table(move(duckdb_result)) {};
 	static ArrowArrayStream *Produce(uintptr_t factory_ptr) {
 		MyPythonTableArrowArrayStreamFactory *factory = (MyPythonTableArrowArrayStreamFactory *)factory_ptr;
-		if (!factory->stream) {
-			auto table_stream = new MyArrowArrayStream(factory->arrow_table.get());
-			factory->stream = &table_stream->stream;
-		}
-		return factory->stream;
+		auto table_stream = new MyArrowArrayStream(factory->arrow_table.get());
```

Comment:
I see a lot of calls to `new` but very few calls to `delete`. Is it necessary to use `new` instead of `unique_ptr`?  If it is, then can we verify that this actually gets cleaned up? I suspect there are a bunch of leaks here, but I don't think the Python tests run with the address sanitizer enabled.